apps/src/studio/studio.js

```diff
@@ -1016,8 +1018,7 @@ function handleActorCollisionsWithCollidableList (
         // Make the projectile/item disappear automatically if this parameter
         // is set:
         if (autoDisappear) {
-          collidable.removeElement();
-          list.splice(i, 1);
+          collidable.beginRemoveElement();
```

Comment:
If you want to use this technique for projectiles as well, you will need to add a function similar to `updateItems` for projectiles so you can remove them when `isGone` returns `true`. Or you can go back to immediately calling `removeElement` for projectiles. You could check to see if the `beginRemoveElement` method exists. But if you do it conditionally, you'll need to flip the loop back to going backwards...